File: sdk/node/examples/test_run_team_tool_kimi.ts
#!/usr/bin/env npx tsx
/**
 * run_team built-in tool smoke test — Kimi model (TypeScript).
 *
 * Tests Session.runTeam() — the typed convenience wrapper that invokes the
 * `run_team` built-in tool (Lead → Worker → Reviewer) directly from SDK code,
 * without going through the Orchestrator API.
 *
 * Compared to test_run_team_kimi.ts (which tests Orchestrator.runTeam()),
 * this test exercises the new Session.runTeam() convenience method added to
 * the Node SDK alongside the RunTeamTool built-in.
 *
 * Run with:
 *   cd sdk/node
 *   npx tsx examples/test_run_team_tool_kimi.ts
 */

import { Agent, Session, ToolResult } from '../index.js';
import * as path from 'path';
import * as fs from 'fs';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const CONFIG_PATH = path.join(
  __dirname, '..', '..', '..', '..', '..', '.a3s', 'config.hcl'
);

function printSection(title: string): void {
  console.log(`\n${'─'.repeat(60)}`);
  console.log(`  ${title}`);
  console.log('─'.repeat(60));
}

async function main(): Promise<void> {
  console.log('=== run_team built-in tool — Kimi integration test (TypeScript) ===\n');

  if (!fs.existsSync(CONFIG_PATH)) {
    console.error(`✗ Config not found: ${CONFIG_PATH}`);
    process.exit(1);
  }
  console.log(`✓ Config: ${CONFIG_PATH}`);

  const agent = await Agent.create(CONFIG_PATH);
  console.log('✓ Agent created (kimi-k2.5)\n');

  // ------------------------------------------------------------------
  // Test 1: session.runTeam() — typed convenience method, all defaults
  // ------------------------------------------------------------------
  printSection('Test 1: session.runTeam() with defaults');

  const session: Session = agent.session('.');
  console.log('Goal: List 3 JavaScript built-in types and describe each briefly.\n');

  const result: ToolResult = await session.runTeam(
    'List 3 JavaScript built-in types and describe each one in one sentence.',
    null, null, null,
    5, // maxSteps
  );

  console.log(`exit_code : ${result.exitCode}`);
  console.log(`output    :\n${result.output}`);

  if (result.exitCode !== 0) throw new Error(`Expected success; got exitCode=${result.exitCode}`);
  if (!result.output.includes('Team run complete')) throw new Error("Missing 'Team run complete' in output");
  console.log('✓ Test 1 passed');

  // ------------------------------------------------------------------
  // Test 2: explicit agent types
  // ------------------------------------------------------------------
  printSection('Test 2: session.runTeam() with explicit agent types');

  const session2: Session = agent.session('.');
  console.log('Goal: Count TypeScript source files in the codebase.\n');

  const result2: ToolResult = await session2.runTeam(
    'Count the number of .ts source files in the codebase and report the total.',
    'general',   // leadAgent
    'explore',   // workerAgent
    'general',   // reviewerAgent
    5,           // maxSteps
  );

  console.log(`exit_code : ${result2.exitCode}`);
  console.log(`output    :\n${result2.output}`);

  if (result2.exitCode !== 0) throw new Error(`Expected success; got exitCode=${result2.exitCode}`);
  if (!result2.output.includes('Team run complete')) throw new Error("Missing 'Team run complete'");
  console.log('✓ Test 2 passed');

  // ------------------------------------------------------------------
  // Test 3: tool() generic call — same underlying route
  // ------------------------------------------------------------------
  printSection("Test 3: session.tool('run_team', {...}) — generic call");

  const session3: Session = agent.session('.');
  const result3: ToolResult = await session3.tool('run_team', {
    goal: 'What programming language is this project primarily written in?',
    max_steps: 10,
  });

  console.log(`exit_code : ${result3.exitCode}`);
  console.log(`output    :\n${result3.output}`);

  if (result3.exitCode !== 0) throw new Error(`Unexpected failure: ${result3.output}`);
  if (!result3.output.includes('Team run complete')) throw new Error("Missing 'Team run complete'");
  console.log('✓ Test 3 passed');

  console.log(`\n${'='.repeat(60)}`);
  console.log('All tests passed ✓');
  console.log('='.repeat(60));
}

main().catch((err: unknown) => {
  console.error('\n✗ Test failed:', err);
  process.exit(1);
});

File: sdk/python/examples/test_run_team_tool_kimi.py
#!/usr/bin/env python3
"""
run_team built-in tool smoke test — Kimi model.

Tests Session.run_team() — the typed convenience wrapper that invokes the
`run_team` built-in tool (Lead → Worker → Reviewer) directly from SDK code,
without going through the Orchestrator API.

Compared to test_run_team_kimi.py (which tests Orchestrator.run_team()),
this test exercises the new Session.run_team() convenience method added to
the Python SDK alongside the RunTeamTool built-in.

Run with:
    cd sdk/python
    python examples/test_run_team_tool_kimi.py
"""

import sys
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from a3s_code import Agent

CONFIG_PATH = os.path.normpath(os.path.join(
    os.path.dirname(__file__),
    "..", "..", "..", "..", "..", ".a3s", "config.hcl"
))


def print_section(title: str) -> None:
    print(f"\n{'─' * 60}")
    print(f"  {title}")
    print(f"{'─' * 60}")


def main() -> None:
    print("=== run_team built-in tool — Kimi integration test (Python) ===\n")

    if not os.path.exists(CONFIG_PATH):
        print(f"✗ Config not found: {CONFIG_PATH}")
        sys.exit(1)
    print(f"✓ Config: {CONFIG_PATH}")

    agent = Agent.create(CONFIG_PATH)
    print("✓ Agent created (kimi-k2.5)\n")

    # ------------------------------------------------------------------
    # Test 1: session.run_team() — typed convenience method, all defaults
    # ------------------------------------------------------------------
    print_section("Test 1: session.run_team() with defaults")

    session = agent.session(".")
    print("Goal: List 3 Python built-in data types and describe each briefly.\n")

    result = session.run_team(
        "List 3 Python built-in data types and describe each one in one sentence.",
        max_steps=5,
    )

    print(f"exit_code : {result.exit_code}")
    print(f"output    :\n{result.output}")

    assert result.exit_code == 0, f"Expected success; got exit_code={result.exit_code}"
    assert "Team run complete" in result.output, "Missing 'Team run complete' in output"
    print("✓ Test 1 passed")

    # ------------------------------------------------------------------
    # Test 2: explicit agent types
    # ------------------------------------------------------------------
    print_section("Test 2: session.run_team() with explicit agent types")

    session2 = agent.session(".")
    print("Goal: Count Rust source files in the codebase.\n")

    result2 = session2.run_team(
        "Count the number of .rs source files in the codebase and report the total.",
        lead_agent="general",
        worker_agent="explore",
        reviewer_agent="general",
        max_steps=5,
    )

    print(f"exit_code : {result2.exit_code}")
    print(f"output    :\n{result2.output}")

    assert result2.exit_code == 0, f"Expected success; got exit_code={result2.exit_code}"
    assert "Team run complete" in result2.output
    print("✓ Test 2 passed")

    # ------------------------------------------------------------------
    # Test 3: tool() generic call — same method, different surface
    # ------------------------------------------------------------------
    print_section("Test 3: session.tool('run_team', {...}) — generic call")

    session3 = agent.session(".")
    result3 = session3.tool("run_team", {
        "goal": "What programming language is this project primarily written in?",
        "max_steps": 10,
    })

    print(f"exit_code : {result3.exit_code}")
    print(f"output    :\n{result3.output}")

    assert result3.exit_code == 0
    assert "Team run complete" in result3.output
    print("✓ Test 3 passed")

    print("\n" + "=" * 60)
    print("All tests passed ✓")
    print("=" * 60)


if __name__ == "__main__":
    main()
